src/Pact/Eval.hs

```diff
@@ -302,16 +328,16 @@ loadModule i@Interface{..} body info gas0 = do
             case dnm of
               Nothing -> return (g, rs)
               Just dn -> do
-                g' <- computeGas (Left (_tInfo t,dn)) (GModuleMember i)
+                g' <- computeGas (Left (_tInfo t,dn)) (GModuleMember (MDInterface i))
                 return (g + g',(dn,t):rs)
       second HM.fromList <$> foldM doDef (gas0,[]) bd
     t -> evalError (_tInfo t) "Malformed interface"
   mapM_ evalUse _interfaceImports
   evaluatedDefs <- evaluateDefs info (fmap (mangleDefs _interfaceName) idefs)
-  let md = ModuleData i evaluatedDefs
+  let md = ModuleData (MDInterface i) evaluatedDefs
   installModule md
   (evalRefs . rsNewModules) %= HM.insert _interfaceName md
-  return gas1
+  return (gas1,_mdModule md)
```

Comment:
I like to force the tuple constructor and make sure there are bangs on the arguments whenever I return a tuple -- more thunk avoidance rules of thumb

Reply:
we also have more of those in here